pkgs/development/haskell-modules/configuration-common.nix

```
@@ -145,6 +145,10 @@ self: super: {
     then addBuildDepend super.halive pkgs.darwin.apple_sdk.frameworks.AppKit
     else super.halive;
 
+  barbly = if pkgs.stdenv.isDarwin
+    then addBuildDepend super.barbly pkgs.darwin.apple_sdk.frameworks.AppKit
+    else super.halive;
```

```suggestion
    else super.barbly;
```

Although since `barbly` only compiles on MacOSX, maybe you could just remove this `if` statement and add the builddepend unconditionally.